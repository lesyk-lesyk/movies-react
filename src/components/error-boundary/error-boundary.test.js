import React from 'react';
import { ErrorBoundary } from './index';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

describe('Error Boundary', () => {
  it('renders', () => {
    const tree = renderer
      .create(<ErrorBoundary>Test</ErrorBoundary>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should display error if error is intercepted during lifecycle hook', done => {
    const wrapper = mount(<ErrorBoundary>Test</ErrorBoundary>);
    wrapper.instance().componentDidCatch();

    process.nextTick(() => {
      wrapper.update();
      expect(wrapper.find('h1').text()).toEqual('Something went wrong :(Please, refresh page.');
      done();
    })
  });
});
