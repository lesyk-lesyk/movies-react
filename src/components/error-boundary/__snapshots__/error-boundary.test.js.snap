// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Boundary renders 1`] = `"Test"`;
